excel-importer/src/main/java/uk/gov/hmcts/ccd/definition/store/excel/validation/SpreadsheetValidator.java

```diff
@@ -74,23 +79,49 @@ private void validateFixedListWorkSheetIsPresent(Map<String, DefinitionSheet> de
             throw new MapperException("A definition must contain a Fixed List worksheet");
     }
 
-
     private void validate(String sheetName, String columnName, SpreadSheetValidationMappingEnum columnNameEnum,
                           String cellValue, String rowNumberInfo) {
         if (columnNameEnum != null) {
             Integer columnMaxLength = columnNameEnum.getMaxLength();
             if (columnMaxLength != null && cellValue.length() > columnMaxLength) {
                 String invalidImportMessage = getImportValidationFailureMessage(sheetName, columnName,
-                                                                                columnMaxLength, rowNumberInfo);
+                    columnMaxLength, rowNumberInfo);
                 throw new InvalidImportException(invalidImportMessage);
             }
         }
     }
 
     public String getImportValidationFailureMessage(String sheetName, String columnName, Integer columnMaxLength,
-                                     String rowNumberInfo) {
+                                                    String rowNumberInfo) {
         return String.format("Invalid Case Definition sheet - In sheet '%s' the column '%s' value should not" +
-                                      " be more than '%s' characters length %s", sheetName,
-                                      columnName, columnMaxLength, rowNumberInfo);
+                " be more than '%s' characters length %s", sheetName,
+            columnName, columnMaxLength, rowNumberInfo);
+    }
+
+    private void validateDisplayContextParameter(Map<String, DefinitionSheet> definitionSheets) {
```

Comment:
this class is growing a lot. Probably better start introducing some helper classes. I'm sure the display context parameter validation is going to grow a lot in the future. Consider extracting this into a separate helper class DisplayContextParameterValidator

Reply:
Also I'm a bit worried of performance. We are looping through all the sheets and all the rows just for the purpose of checking the display context parameter. If we do this looping for any column in the future where we need to add validation then perf of the import will start to suffer. 
Ideally we should design the import to do the looping once and then apply all the various sorts of validations

Reply:
It will be better to report consolidated list of all errors from all tabs. Instead of throwing the exception immediately. 
You can look into the implementation done in below 

https://github.com/hmcts/ccd-definition-store-api/blob/RDM-7627-Validation-For-Table/domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/service/casetype/CaseTypeServiceImpl.java#L115

Reply:
@kiran-yenigala-hmcts I would not add that code to the Parser. Parser is already very complicated class and should only do parsing.
Can't we have this as an additional implementation of the already  existing interface `DisplayContextParameterValidator`?